test/integration/targets/zabbix_hostmacro/tasks/main.yml

```diff
@@ -0,0 +1,165 @@
+      assert:
+        that: not zbxhmacro_existing is changed
+
+    - name: test - attempt to create same host macro in zabbix native format
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: "{$ZBXHMACRO_TEST01}"
+        macro_value: 123
+      register: zbxhmacro_existing_native
+
+    - name: assert that nothing has been changed
+      assert:
+        that: not zbxhmacro_existing_native is changed
+
+    - name: test - attempt to create new host macro in zabbix native format
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: "{$ZBXHMACRO_TEST02}"
+        macro_value: abcd
+      register: zbxhmacro_new_native
+
+    - name: assert that nothing macro was created
+      assert:
+        that: zbxhmacro_new_native is changed
+
+    - name: test - attempt to update host macro with string value while force=no
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: zbxhmacro_test01
+        macro_value: abc
+        force: no
+      register: zbxhmacro_update_noforce
+
+    - name: assert that nothing has been changed
+      assert:
+        that: not zbxhmacro_update_noforce is changed
+
+    - name: test - attempt to update host macro with string value
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: zbxhmacro_test01
+        macro_value: abc
+      register: zbxhmacro_update
+      ignore_errors: True
+
+    - name: assert that host macro was updated
+      assert:
+        that: zbxhmacro_update is changed
+
+    - name: test - attempt to create host macro with context
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: LOW_SPACE_LIMIT:/home
+        macro_value: 10
+      register: zbxhmacro_context_new
+
+    - name: assert that macro was created
+      assert:
+        that: zbxhmacro_context_new is changed
+
+    - name: test - attempt to create same host macro with context
+      zabbix_hostmacro:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        host_name: zbx_hmacro_host01
+        macro_name: LOW_SPACE_LIMIT:/home
```

Comment:
I think need a test where `LOW_SPACE_LIMIT` is lowercase, what do you think?
https://github.com/ansible/ansible/blob/011e0176c2930ecff6dbfb2e0b8bca6c5f58b1bd/lib/ansible/modules/monitoring/zabbix/zabbix_hostmacro.py#L180
(Is it unnecessary because it is guaranteed by the test of https://github.com/ansible/ansible/pull/66033/files#diff-593491eb48b2f5d92429d3ec52ceb91aR17?)

Reply:
You have a good point. We should test context macros with lower case as well due to a hidden 'split' mechanic. Will add it!

Reply:
Decided to change it to lower case in the previous task and let it be upper case in this one. In this way it ensures that module converted macro to uppercase before comparing/adding/updating it.